### PR TITLE
Feature improve relayer error handling

### PR DESCRIPTION
**Pull request by juniuszhou**
Relayer just print error and quit if any exceptional case before. Add better error handling for relayer.

### CODE REVIEW

File: cmd/ebrelayer/txs/signature.go [hunk 1]
```diff
@@ -21,13 +22,15 @@ func LoadPrivateKey() (key *ecdsa.PrivateKey, err error) {
 	// Private key for validator's Ethereum address must be set as an environment variable
 	rawPrivateKey := os.Getenv("ETHEREUM_PRIVATE_KEY")
 	if strings.TrimSpace(rawPrivateKey) == "" {
-		log.Fatal("Error loading ETHEREUM_PRIVATE_KEY from .env file")
+		log.Println("Error loading ETHEREUM_PRIVATE_KEY from .env file")
```
Comment (utx0) [inline cmd/ebrelayer/txs/signature.go:25]: Can we just change this line too:

```
 log.Println("Error loading ETHEREUM_PRIVATE_KEY") 
```
As we wont be using `.env` files in all deployments

File: cmd/ebrelayer/txs/signature.go [hunk 2]
```diff
@@ -21,13 +22,15 @@ func LoadPrivateKey() (key *ecdsa.PrivateKey, err error) {
...
+		return &ecdsa.PrivateKey{}, errors.New("can't load ethereum private key from .env file")
 	}
 
 	// Parse private key
 	privateKey, err := crypto.HexToECDSA(rawPrivateKey)
 	if err != nil {
-		log.Fatal(err)
+		log.Println(err)
+		return privateKey, err
```
Comment (utx0) [inline cmd/ebrelayer/txs/signature.go:33]: I feel like this line should be:
```
return nil, err
```
due to the function signiture using a pointer.

File: cmd/ebrelayer/txs/relayToEthereum.go [hunk 3]
```diff
@@ -102,27 +109,34 @@ func RelayOracleClaimToEthereum(provider string, contractAddress common.Address,
...
 // initRelayConfig set up Ethereum client, validator's transaction auth, and the target contract's address
 func initRelayConfig(provider string, registry common.Address, event types.Event, key *ecdsa.PrivateKey,
-) (*ethclient.Client, *bind.TransactOpts, common.Address) {
+) (*ethclient.Client, *bind.TransactOpts, common.Address, error) {
 	// Start Ethereum client
 	client, err := ethclient.Dial(provider)
 	if err != nil {
-		log.Fatal(err)
+		log.Println(err)
+		return &ethclient.Client{}, &bind.TransactOpts{}, common.Address{}, err
```
Comment (utx0) [inline cmd/ebrelayer/txs/relayToEthereum.go:117]: I feel like this line should be:
```
return nil, nil, common.Address{}, err
```
due to the function signiture using ponters

File: cmd/ebrelayer/txs/relayToEthereum.go [hunk 4]
```diff
@@ -102,27 +109,34 @@ func RelayOracleClaimToEthereum(provider string, contractAddress common.Address,
...
 	}
 
 	// Load the validator's address
 	sender, err := LoadSender()
 	if err != nil {
-		log.Fatal(err)
+		log.Println(err)
+		return &ethclient.Client{}, &bind.TransactOpts{}, common.Address{}, err
```
Comment (utx0) [inline cmd/ebrelayer/txs/relayToEthereum.go:124]: same here;

I feel like this line should be:
```
return nil, nil, common.Address{}, err
```
due to the function signiture using ponters

File: cmd/ebrelayer/txs/relayToEthereum.go [hunk 5]
```diff
@@ -102,27 +109,34 @@ func RelayOracleClaimToEthereum(provider string, contractAddress common.Address,
...
+
 	}
 
 	nonce, err := client.PendingNonceAt(context.Background(), sender)
 	if err != nil {
-		log.Fatal(err)
+		log.Println(err)
+		return &ethclient.Client{}, &bind.TransactOpts{}, common.Address{}, err
```
Comment (utx0) [inline cmd/ebrelayer/txs/relayToEthereum.go:131]: same here:

I feel like this line should be:
```
return nil, nil, common.Address{}, err
```
due to the function signiture using ponters

File: cmd/ebrelayer/txs/relayToEthereum.go [hunk 6]
```diff
@@ -102,27 +109,34 @@ func RelayOracleClaimToEthereum(provider string, contractAddress common.Address,
...
+
 	}
 
 	gasPrice, err := client.SuggestGasPrice(context.Background())
 	if err != nil {
-		log.Fatal(err)
+		log.Println(err)
+		return &ethclient.Client{}, &bind.TransactOpts{}, common.Address{}, err
```
Comment (utx0) [inline cmd/ebrelayer/txs/relayToEthereum.go:138]: same here:

I feel like this line should be:
```
return nil, nil, common.Address{}, err
```
due to the function signiture using ponters

File: cmd/ebrelayer/txs/relayToEthereum.go [hunk 7]
```diff
@@ -147,7 +161,9 @@ func initRelayConfig(provider string, registry common.Address, event types.Event
 	// Get the specific contract's address
 	target, err := GetAddressFromBridgeRegistry(client, registry, targetContract)
 	if err != nil {
-		log.Fatal(err)
+		log.Println(err)
+		return &ethclient.Client{}, &bind.TransactOpts{}, common.Address{}, err
```
Comment (utx0) [inline cmd/ebrelayer/txs/relayToEthereum.go:165]: and here;

I feel like this line should be:
```
return nil, nil, common.Address{}, err
```
due to the function signiture using ponters

**Review by utx0 — CHANGES_REQUESTED** [review]
Some minor fixes needed

File: cmd/ebrelayer/txs/relayToEthereum.go (hunk 7 above)
Comment (juniuszhou) [reply cmd/ebrelayer/txs/relayToEthereum.go:165]: Accepted.

File: cmd/ebrelayer/txs/relayToEthereum.go (hunk 6 above)
Comment (juniuszhou) [reply cmd/ebrelayer/txs/relayToEthereum.go:138]: Accepted.

File: cmd/ebrelayer/txs/relayToEthereum.go (hunk 5 above)
Comment (juniuszhou) [reply cmd/ebrelayer/txs/relayToEthereum.go:131]: Accepted.

File: cmd/ebrelayer/txs/relayToEthereum.go (hunk 4 above)
Comment (juniuszhou) [reply cmd/ebrelayer/txs/relayToEthereum.go:124]: Accepted.

File: cmd/ebrelayer/txs/relayToEthereum.go (hunk 3 above)
Comment (juniuszhou) [reply cmd/ebrelayer/txs/relayToEthereum.go:117]: Accepted.

File: cmd/ebrelayer/txs/signature.go (hunk 2 above)
Comment (juniuszhou) [reply cmd/ebrelayer/txs/signature.go:33]: Accepted.

File: cmd/ebrelayer/txs/signature.go (hunk 1 above)
Comment (juniuszhou) [reply cmd/ebrelayer/txs/signature.go:25]: Yes, .env file just for local machine, will remove it.

**Review by utx0 — APPROVED** [review]
Very nice! Thank you.

**Pull request merged**